pkg/pprof/merge_test.go

```diff
@@ -19,6 +19,30 @@ func Test_Merge_Single(t *testing.T) {
 	testhelper.EqualProto(t, p.Profile, m.Profile())
 }
 
+func Fuzz_Merge_Single(f *testing.F) {
+
+	for _, file := range []string{
+		"testdata/go.cpu.labels.pprof",
+		"testdata/heap",
+		"testdata/profile_java",
+		"testdata/profile_rust",
+	} {
+		raw, err := OpenFile(file)
+		require.NoError(f, err)
+		data, err := raw.MarshalVT()
+		require.NoError(f, err)
+		f.Add(data)
+	}
+
+	f.Fuzz(func(t *testing.T, data []byte) {
+		var p profilev1.Profile
+		if err := p.UnmarshalVT(data); err != nil {
```

Comment:
As @aleks-p has pointed out yesterday, we exit early.

```suggestion
		if err := p.UnmarshalVT(data); err == nil {
```

Reply:
Oops, sorry about that! 😅 I made sure I was testing the correct version. I ran it again for 2 hours and didn't spot any errors. Without the recent changes, it fails very quickly